src/reporters/WebPageTestReporter.ts

```diff
@@ -0,0 +1,164 @@
+
+      return {
+        formatScore,
+        name: name,
+        scores: {
+          firstView: firstView[key],
+          repeatView: repeatView[key],
+        },
+      };
+    });
+  }
+
+  async getReports(): Promise<Report[]> {
+    const runTestResponse = await bluebird.fromNode<RunTestResponse>(cb => {
+      this.wpt.runTest(
+        this.url,
+        {
+          connectivity: '3G',
+          location: 'Dulles_MotoG4',
+          lighthouse: true,
+          pageSpeed: true,
+          keepOriginalUserAgent: true,
+          pollResults: 5,
+        },
+        cb,
+      );
+    });
+
+    if (
+      // tslint:disable-next-line use-default-type-parameter
+      isSuccessfulResponse<any>(runTestResponse) &&
+      isWaitUntilTestCompleteResponse(runTestResponse)
+    ) {
+      const { data } = runTestResponse;
+
+      const reports: Report[] = [
+        {
+          name: 'WebPageTest',
+          url: data.summary,
+          records: [
+            {
+              name: 'Number of requests',
+              scores: {
+                firstView: getNumberOfRequests(data.median.firstView),
+                repeatView: getNumberOfRequests(data.median.repeatView),
+              },
+              formatScore: defaultFormat,
+            },
+            {
+              name: 'Time to first byte',
+              scores: {
+                firstView: data.median.firstView.TTFB,
+                repeatView: data.median.repeatView.TTFB,
+              },
+              formatScore: formatMillisecondsAsSeconds,
+            },
+            {
+              name: 'Fully loaded',
+              scores: {
+                firstView: data.median.firstView.fullyLoaded,
+                repeatView: data.median.repeatView.fullyLoaded,
+              },
+              formatScore: formatMillisecondsAsSeconds,
+            },
+            {
+              name: 'Response size',
+              scores: {
+                firstView: data.median.firstView.bytesIn,
+                repeatView: data.median.repeatView.bytesIn,
+              },
+              formatScore: formatBytesAsKibibytes,
+            },
+            {
+              name: 'Response size (compressed)',
+              scores: {
+                firstView: data.median.firstView.gzip_total,
+                repeatView: data.median.repeatView.gzip_total,
+              },
+              formatScore: formatBytesAsKibibytes,
+            },
+          ],
+        },
+      ];
+
+      if (hasLighthouseData(runTestResponse)) {
+        reports.push({
+          name: 'Lighthouse via WebPageTest',
+          records: WebPageTestReporter.getLighthouseRecords(
+            runTestResponse.data,
+          ),
+        });
+      } else {
+        // tslint:disable-next-line:no-multiline-string
+        console.warn(oneLine`
+          Could not find Lighthouse data in WPT API response.
+          Does the location specified support Lighthouse?
```

Comment:
What is meant by `location specified` here?

Reply:
One of www.webpagetest.org/getLocations.php?f=html. I should add this to the message.